src/pages/vaults/VaultDepositWithdrawForm.tsx

```diff
@@ -0,0 +1,637 @@
+import { FormInput } from '@/components/FormInput';
+import { FormMaxInputToggleButton } from '@/components/FormMaxInputToggleButton';
+import { Icon, IconName } from '@/components/Icon';
+import { InputType } from '@/components/Input';
+import { Link } from '@/components/Link';
+import { Output, OutputType } from '@/components/Output';
+import { WithDetailsReceipt } from '@/components/WithDetailsReceipt';
+
+import { calculateCanViewAccount, calculateIsAccountViewOnly } from '@/state/accountCalculators';
+import { getSubaccount } from '@/state/accountSelectors';
+import { useAppSelector } from '@/state/appTypes';
+import { getUserVault } from '@/state/vaultSelectors';
+
+import { MustBigNumber } from '@/lib/numbers';
+
+const renderDiffOutput = ({
+  type,
+  value,
+  newValue,
+  showSign,
+  withDiff,
+}: Pick<
+  Parameters<typeof DiffOutput>[0],
+  'type' | 'value' | 'newValue' | 'withDiff' | 'showSign'
+>) => (
+  <DiffOutput
+    type={type}
+    value={value}
+    newValue={newValue}
+    showSign={showSign}
+    withDiff={withDiff}
+  />
+);
+
+type VaultFormError = {
+  type: AlertType;
+  key: string;
+  short?: string;
+  long?: React.ReactNode;
+};
+
+// errors we don't want to show aggressive visual cues about, just disable submit
+const lightErrorKeys = new Set(['disconnected', 'view-only', 'amount-empty']);
+
+const SLIPPAGE_PERCENT_WARN = 0.01;
+const SLIPPAGE_PERCENT_ACK = 0.01;
+
+type VaultDepositWithdrawFormProps = {
+  initialType?: 'deposit' | 'withdraw';
+  onSuccess?: () => void;
+};
+
+export const VaultDepositWithdrawForm = ({
+  initialType,
+  onSuccess,
+}: VaultDepositWithdrawFormProps) => {
+  const stringGetter = useStringGetter();
+  const { vaultsLearnMore } = useURLConfigs();
+  const isAccountViewOnly = useAppSelector(calculateIsAccountViewOnly);
+  const canViewAccount = useAppSelector(calculateCanViewAccount);
+
+  const { userBalance } = useAppSelector(getUserVault) ?? {};
+  const { freeCollateral, marginUsage } = useAppSelector(getSubaccount) ?? {};
+
+  const [selectedType, setSelectedType] = useState<'deposit' | 'withdraw'>(
+    initialType ?? 'deposit'
+  );
+  const [amount, setAmountState] = useState('');
+  const [isSubmitting] = useState(false);
+  const [currentForm, setCurrentForm] = useState<'input' | 'confirm'>('input');
+  const [slippageAck, setSlippageAck] = useState(false);
+
+  const slippagePercent = 0.05;
+  const estimatedWithdrawalAmount = MustBigNumber(amount).times(1 - slippagePercent);
+  const freeCollateralUpdated =
+    selectedType === 'deposit'
+      ? MustBigNumber(MustBigNumber(freeCollateral?.current).minus(amount).toFixed(2)).toNumber()
+      : MustBigNumber(freeCollateral?.current).plus(estimatedWithdrawalAmount).toNumber();
+  const marginUsageUpdated =
+    selectedType === 'deposit'
+      ? MustBigNumber(marginUsage?.current).minus(0.05).toNumber()
+      : MustBigNumber(marginUsage?.current).plus(0.05).toNumber();
```

Comment:
should this be `slippagePercent` ? might be nice to extract this into a const as well just to understand where this number comes from or what its supposed to denote

Reply:
this is just temporary, made up number until we have abacus computing it for real.